### ISSUE TITLE
Incorrect error when using vkGetSemaphoreCounterValue() to check command buffer completion

### ISSUE THREAD

**Issue by mbechard** [LunarG/CrashDiagnosticLayer#173]
**Environment:**
 - OS:  Windows 11
 - GPU and driver version: RTX 6000 Ada, 572.93
 - SDK or header version if building from repo: 1.4.309.0
 - Options enabled (synchronization, best practices, etc.): Crash Diagnostics configuration

**Describe the Issue**

I'm checking to see if a command buffer is ready to be reset with call to `vkGetSemaphoreCounterValue()` for a timeline semaphore it is signaling. The Crash Diagnostic configuration produces the error when I then try to reset it:
```
ERROR : GENERAL - Message Id Number: 1 | Message Id Name: CDL
        Invalid Command Buffer Usage: Reset of VkCommandBuffer in use by GPU: uniqueID: 11, GX_Device::buildDefaultAttribBuffers: Primary Gfx (0x000001CFA60D6B40)
```

This error is not produced when using Validation configuration. If I instead use a wait() with a 0 timeout, the error is avoided.

You can reproduce the error by turning on the Crash Diagnostics and having it write to a log file, and then run TouchDesigner from here:
https://download.derivative.ca/TouchDesigner.2023.12230.exe

**Comment by StefanPoelloth** [comment]
I have encountered the same problem when using QueueWaitIdle to wait for completion:
- AllocateCommandBuffers & BeginCommandBuffer
- record some commands
- EndCommandBuffer
- QueueSubmit
- QueueWaitIdle
- FreeCommandBuffers
